xcat-inventory/xcclient/inventory/inventorydiff.py

```diff
@@ -157,48 +156,107 @@ def get_diff_string(self, diff_dict):
 
 class InventoryDiff(object):
 
-    def __init__(self, obj1, obj2, srctype='other', objtype='f'):
-        self.obj1 = obj1
-        self.obj2 = obj2
-        self.srctype = srctype
+    def __init__(self, objs, objtype, isgit=False):
         self.objtype = objtype
+        if objtype == 'f':
+            self.obj1 = objs.pop(0)
+        else:
+            self.obj1 = 'xCAT_DB'
+        self.obj2 = objs.pop(0)
+        self.isgit = isgit
         self.fmt = 'json'
 
-    def _get_data(self):
-        try:
-            d1, self.fmt = utils.loadfile(filename=self.obj1)
-            d2, self.fmt = utils.loadfile(filename=self.obj2)
-        except Exception,e:
-            (retcode,out,err)=utils.runCommand("diff -u %s %s"%(self.obj1,self.obj2))
-            if out:
-                out=re.sub(r"%s.*"%(self.obj1),self.obj2,out)
-                out=re.sub(r"%s.*"%(self.obj2),self.obj2,out)
-            if err:
-                err=re.sub(r"%s.*"%(self.obj1),self.obj2,err)
-                err=re.sub(r"%s.*"%(self.obj2),self.obj2,err)
-            return 1, out, err
-        return 0, d1, d2
+    def _get_file_data(self, data_file):
+        data, self.fmt = utils.loadfile(filename=data_file)
+        return data
+
+    def _diff_with_cmd(self):
```

Comment:
suggest split the code logic to be 2 subroutines, one for dict comparison(return the diff result as a dict) and the other for display of comparison result, so that it can be reused in other places in xcat-inventory